### PR TITLE
Encoder training: new regimes and specs

### PR DESCRIPTION
**Pull request by nkemnitz**
* includes / requires #446 
* includes / requires #498 
* includes / requires #513 
* includes / requires #514 
* requires #537

### CODE REVIEW

File: zetta_utils/__init__.py [hunk 1]
```diff
@@ -61,3 +61,6 @@ def load_training_modules():
     )
     from .layer import volumetric
     from .layer.volumetric import cloudvol
+
+
+load_all_modules()
```
Comment (supersergiy) [inline zetta_utils/__init__.py:66]: This should be `try_load_train_inference`

**Review by supersergiy — CHANGES_REQUESTED** [review]
Hopefully the `.py` specs won't be too intimidating for whoever has to pick this up! 
Let's stick with `.cue` in the future for now...

File: zetta_utils/__init__.py (hunk 1 above)
Comment (nkemnitz) [reply zetta_utils/__init__.py]: Thanks! Fixed.

**Review by supersergiy — APPROVED** [review]
💯

File: zetta_utils/training/lightning/train.py [hunk 2]
```diff
@@ -84,55 +82,82 @@ def lightning_train(
...
+            trainer=trainer if not isinstance(trainer, dict) else builder.build(trainer),
+            train_dataloader=train_dataloader
+            if not isinstance(train_dataloader, dict)
+            else builder.build(train_dataloader),
+            val_dataloader=val_dataloader
+            if not isinstance(val_dataloader, dict)
+            else builder.build(val_dataloader),
+            full_state_ckpt_path=full_state_ckpt_path,
+        )
+    else:
```
Comment (akhileshh) [inline zetta_utils/training/lightning/train.py:104]: there are no statements after the else so we can return and remove this block, one less level of indentation

File: zetta_utils/training/lightning/train.py [hunk 3]
```diff
@@ -267,11 +294,29 @@ def _lightning_train_remote(
...
+            trainer_devices = train_args["trainer"]["devices"]
+            if (
+                isinstance(trainer_devices, int)
+                and trainer_devices != -1
+                and trainer_devices != num_devices
+            ):
+                raise ValueError(
+                    f"Trainer specification uses {trainer_devices} devices, "
+                    f"while `nvidia.com/gpu` limit is {num_devices}."
+                )
```
Comment (akhileshh) [inline zetta_utils/training/lightning/train.py:310]: This check applies to `num_nodes=1` as well, rest looks good

**Review by supersergiy — APPROVED** [review]

File: zetta_utils/training/lightning/train.py [hunk 4]
```diff
@@ -267,13 +293,31 @@ def _lightning_train_remote(
     Creates a volume mount for `train.cue` in `/opt/zetta_utils/specs`.
     Runs the command `zetta run specs/train.cue` on one or more worker pods.
     """
+    if train_args["trainer"]["accelerator"] == "gpu":
```
Comment (torms3) [inline zetta_utils/training/lightning/train.py:296]: The [[example DDP spec]](https://github.com/ZettaAI/zetta_utils/blob/9ddf8818fede27a119f5912ba17ecbfcc1220a40/specs/examples/training/ddp.cue#L63) sets `accelerator` to "cuda", resulting in a `NotImplementedError`.

**Pull request merged**